File: Reroll.Web/Reroll.Web/ClientApp/src/models/WeaponModel.ts
﻿
     import { DiceTypeEnum } from './dice-type-enum';

export class Weapon   {
	public name: string;
	public attackBonus: number;
	public diceCount: number;
	public diceType: DiceTypeEnum;
	public critical: string;
	public notesList: string[];
	public isRanged: boolean;
}

File: Reroll.Models/WeaponModel.cs
using System.Collections.Generic;

namespace Reroll.Models
{
    public class WeaponModel
    {
        public string Name { get; set; }
        public int AttackBonus { get; set; }
        public int DiceCount { get; set; }
        public DiceTypeEnum DiceType { get; set; }
        public string Critical { get; set; }
        public List<string> NotesList { get; set; }
        public bool IsRanged { get; set; }
    }
}